src/overlays/actors/ovl_En_Poh/z_en_poh.c

```diff
@@ -20,111 +53,1095 @@ const ActorInit En_Poh_InitVars = {
+        func_80ADE6D4(this);
+    } else if (this->unk_198 == 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 10.0f, 10.0f, 10.0f, 4);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Poh/func_80ADEC9C.s")
+void func_80ADFA90(EnPoh* this, s32 arg1) {
+    f32 multiplier;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Poh/func_80ADEDFC.s")
+    this->lightColor.a = CLAMP(this->lightColor.a + arg1, 0, 255);
+    if (arg1 < 0) {
+        multiplier = this->lightColor.a * 0.003921569f;
+        this->actor.scale.z = 0.0056000002f * multiplier + 0.00140000006f;
+        this->actor.scale.x = 0.0056000002f * multiplier + 0.00140000006f;
+        this->actor.scale.y = (0.007f - 0.007f * multiplier) + 0.007f;
+    } else {
+        multiplier = 1.0f;
+        this->actor.scale.z = this->lightColor.a * 2.7450982e-05f;
+        this->actor.scale.y = this->lightColor.a * 2.7450982e-05f;
+        this->actor.scale.x = this->lightColor.a * 2.7450982e-05f;
+        this->actor.posRot.pos.y = this->actor.initPosRot.pos.y + 0.05882353f * this->lightColor.a;
+    }
+    this->lightColor.r = this->info->lightColor.r * multiplier;
+    this->lightColor.g = this->info->lightColor.g * multiplier;
+    this->lightColor.b = this->info->lightColor.b * multiplier;
+    Lights_PointNoGlowSetInfo(&this->lightInfo, this->actor.posRot.pos.x, this->actor.posRot.pos.y,
+                              this->actor.posRot.pos.z, this->info->lightColor.r, this->info->lightColor.g,
+                              this->info->lightColor.b, this->lightColor.a * 0.78431373f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Poh/func_80ADEECC.s")
+void func_80ADFE28(EnPoh* this, GlobalContext* globalCtx) {
+    this->actor.initPosRot.pos.y += 2.0f;
+    func_80ADFA90(this, 20);
+    if (this->lightColor.a == 255) {
+        EnPoh_Talk(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Poh/func_80ADEF38.s")
+void func_80ADFE80(EnPoh* this, GlobalContext* globalCtx) {
+    DECR(this->unk_198);
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        if (this->actor.params >= EN_POH_SHARP) {
+            func_80ADE9BC(this);
+        } else {
+            func_80ADE998(this);
+        }
+        return;
+    }
+    if (this->unk_198 == 0) {
+        func_80ADE950(this, 1);
+        this->actor.flags &= ~0x10000;
+        return;
+    }
+    if (this->colliderCyl.base.maskA & 2) {
+        this->actor.flags |= 0x10000;
+        func_8002F2F4(&this->actor, globalCtx);
+    } else {
+        this->actor.flags &= ~0x10000;
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderCyl.base);
+    }
+    this->actor.posRot.pos.y = Math_Sins(this->unk_195 * 0x800) * 5.0f + this->actor.initPosRot.pos.y;
+    DECR(this->unk_195);
```

Comment:
```suggestion
    if (this->unk_195 != 0) {
        this->unk_195--;
    }
```

Reply:
Expanding DECR in this way doesn't match unfortunately

Reply:
Even if you add a `if (0) {}`?

Reply:
`if (0)` and similar didn't work but `-= 1` over `--` worked.